Client 3772 c expr index examples

sfalvo-aerospike opened the pull request:
Contribute expression index query examples, tracking the same documentation at https://aerospike.com/docs/database/learn/tutorials/quick-start-expression-index/#create-expression-index-and-query .  This code serves as the basis for the C code examples on that page (in a separate PR in the aerospike-websites repo).  

Note that I'm opening this PR to get feedback early.  It was tested in the absence of an index, but remains to be tested with an index.  I figure if I need to debug code, then that'd be a perfect time to address code review feedback as well.

sfalvo-aerospike committed:
CLIENT-3772: Print Base64 repr of index expression

sfalvo-aerospike committed:
CLIENT-3772: Example of inserting data, some not indexed

sfalvo-aerospike committed:
CLIENT-3772: Add command to remove records

sfalvo-aerospike committed:
CLIENT-3772: Print out the results of a query

sfalvo-aerospike committed:
CLIENT-3772: Remove dead code

sfalvo-aerospike committed:
CLIENT-3772: Fix GET Example comment to be more descriptive

BrianNichols reviewed (changes requested):
In Makefile, the following line should be in alphabetical order.

`$(MAKE) -C expr-index $@`

sfalvo-aerospike committed:
CLIENT-3772: Sort makefile alphabetically

sfalvo-aerospike committed:
CLIENT-3772: Fix expression

Expression previously used was known bad, the work-around is to use variadic
OR against one or more EQ expressions.

sfalvo-aerospike commented:
> In Makefile, the following line should be in alphabetical order.
> 
> `$(MAKE) -C expr-index $@`

Fixed in 2nd latest commit

sfalvo-aerospike committed:
CLIENT-3772: First steps at making code examples self-running.
